src/elements/common/current-user/withCurrentUser.tsx

```diff
@@ -0,0 +1,142 @@
+         *
+         * @private
+         * @param {Error} error - Error object
+         * @param {Error} code - the code for the error
+         * @param {Object} contextInfo - the context info for the error
+         * @return {void}
+         */
+        errorCallback = (error: ElementsXhrError, code: string, contextInfo: Object = {}): void => {
+            /* eslint-disable no-console */
+            console.error(error);
+            /* eslint-enable no-console */
+
+            // eslint-disable-next-line react/prop-types
+            this.props.onError(error, code, contextInfo);
+        };
+
+        /**
+         * Fetches a Users info
+         *
+         * @private
+         * @param {User} [user] - Box User. If missing, gets user that the current token was generated for.
+         * @param {boolean} shouldDestroy
+         * @return {void}
+         */
+        fetchCurrentUser = (user?: User, shouldDestroy = false): void => {
+            const { api, file } = this.props;
+
+            if (!file) {
+                throw getBadItemError();
+            }
+
+            if (typeof user === 'undefined') {
+                api.getUsersAPI(shouldDestroy).getUser(
+                    file.id,
+                    this.fetchCurrentUserSuccessCallback,
+                    this.fetchCurrentUserErrorCallback,
+                );
+            } else {
+                this.setState({ currentUser: user, currentUserError: undefined });
+            }
+        };
+
+        /**
+         * User fetch success callback
+         *
+         * @private
+         * @param {Object} currentUser - User info object
+         * @return {void}
+         */
+        fetchCurrentUserSuccessCallback = (currentUser: User): void => {
+            this.setState({ currentUser, currentUserError: undefined });
+        };
+
+        /**
+         * Handles a failed file user info fetch
+         *
+         * @private
+         * @param {ElementsXhrError} e - API error
+         * @return {void}
+         */
+        fetchCurrentUserErrorCallback = (e: ElementsXhrError, code: string) => {
+            this.setState({
+                currentUser: undefined,
+                currentUserError: {
+                    maskError: {
+                        errorHeader: messages.currentUserErrorHeaderMessage,
+                        errorSubHeader: messages.defaultErrorMaskSubHeaderMessage,
+                    },
+                },
+            });
+
+            this.errorCallback(e, code, {
+                error: e,
+            });
+        };
+
+        render() {
+            const { currentUser, currentUserError } = this.state;
+            return <WrappedComponent {...this.props} currentUser={currentUser} currentUserError={currentUserError} />;
+        }
+    }
+
+    const displayName = WrappedComponent.displayName || WrappedComponent.name || 'Component';
+    ComponentWithCurrentUser.displayName = `WithCurrentUser(${displayName})`;
```

Comment:
Should this use a lower-case `w`?

Reply:
I was following this https://github.com/box/box-ui-elements/blob/d0d0132795f84329df95b002620b4d0ace21f51a/src/elements/common/annotator-context/withAnnotations.tsx#L184 do you want me to change it to a lower case ?

Reply:
If that's how we have it elsewhere, let's leave it be. It's not super important.